src/libraries/System.Runtime.Extensions/tests/System/AppDomainTests.cs

```diff
@@ -639,6 +639,32 @@ public void AssemblyResolve_IsNotCalledForCoreLibResources()
             }).Dispose();
         }
 
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/43909", TestRuntimes.Mono)]
+        public void AssemblyResolve_ExceptionPropagatesCorrectly()
+        {
+            RemoteExecutor.Invoke(() => {
+                bool handlerExceptionThrown = false;
+
+                AppDomain.CurrentDomain.AssemblyResolve += (sender, e) =>
+                {
+                    throw new Exception("Failure");
+                };
+
+                try
+                {
+                    Type.GetType("Some.Assembly.Type, Some.Assembly");
+                }
+                catch (Exception e)
+                {
+                    Assert.NotNull(e);
```

Comment:
Can you `throw null`? what is this assertion asserting?

It would be better to check for what we actually expect:

```
   catch (FileLoadException e) {   // note FLE not a generic Exception
     Assert.IsNotNull (e.InnerException);
     Assert.Equals ("Failure", e.InnerException.Message);
   }
```